### DISCUSSION TITLE
Create DB and document types with a script

### DISCUSSION THREAD

**Discussion by vic0824** [ArcadeData/arcadedb#252] [Q&A]
I'm assessing the feasibility of migrating a project from Hsqldb to arcadedb. I'd like to maintain the capability of having a simple script that can be executed to automatically create the DB and the necessary document types, without starting the DB server. With Hsqldb, I can create such a script that uses the console and passes an SQL script containing all the necessary commands.
With ArcadeDB, it has to be done differently, because in order to create DB objects, the server must be started.
This is not a big deal, as the same script could start the server and create the DB if it doesn't exist.
However, I can't find a way to put all the DB creation commands in a file that is easily editable.
The manual describes several ways to create a DB and its document types:
1 - using Studio: this requires issuing the individual commands manually
2 - using the console: this requires issuing the individual commands manually
3 - programmatically, writing a Java application: this would require re-compilation if some DB characteristics have to be modified
4 - using HTTP/JSON Protocol: this would introduce a dependency on some sort of http client (e.g. CURL), or it would require to write an html file that contains the AJAX calls and load it from a browser, which seems an ugly solution
Are there other alternatives?

**Comment by tolgaulas** [comment]
Hello, we use flat text file stacked command and some scripting to call the commands as in your point #4,
over html api. Not so ugly frankly from our side.

**Comment by lvca** [comment]
@vic0824 you can use ArcadeDB as embedded without starting a server. Run the console and create a database with a local path.
